### DISCUSSION TITLE
Declare specific triplet for different dependencies in manifest

### DISCUSSION THREAD

**Discussion by niyue** [microsoft/vcpkg#20571] [Ideas]
Currently in vcpkg manifest mode, it provides the capability to specify triplet using environment variable `VCPKG_TARGET_TRIPLET` or command line parameter `--triplet=<triplet>`. If users would like to specify different triplet for different dependencies, for example, dep1/dep2 need to be built with debug mode, and dep3/dep4 need to be built with both debug/release mode, the only option is to draft a complex triplet with some conditional logic like `x64-linux-partial-debug-partial-both-debug-and-release.cmake` to make it work. Any change to the triplet logic will invalidate binary caching, which is not desirable. 
```
if(PORT MATCHES "dep1-")
    set(VCPKG_BUILD_TYPE debug)
endif()

if(PORT MATCHES "dep2-")
    set(VCPKG_BUILD_TYPE debug)
endif()
......
```

I am wondering if there is any chance we could add the triplet as part of the dependency declaration in manifest file, so that new triplet can be added for some dependencies without re-building existing dependencies.
```json
{
  "dependencies": [
     {
        "name": "dep1",
        "triplet": "x64-linux-debug"
     },
     {
        "name": "dep2",
        "triplet": "x64-linux-debug"
     },
    "dep3", 
    "dep4"
  ]
}
```

**Comment by Neumann-A** [comment]
This is basically #19984
